### ISSUE TITLE
npm cannot find @npmcorp/copy

### ISSUE THREAD

**Issue by mbartelsm** [npm/npm#16037]
#### What's going wrong?

I'm unable to install any package

Verbose:

````
npm info it worked if it ends with ok
npm verb cli [ 'C:\\Program Files\\nodejs\\node.exe',
npm verb cli   'C:\\Users\\mbart\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
npm verb cli   'install',
npm verb cli   'qawgehrsfgrw',
npm verb cli   '--verbose' ]
npm info using npm@4.4.2
npm info using node@v7.7.2
npm verb stack Error: Cannot find module '@npmcorp/copy'
npm verb stack     at Function.Module._resolveFilename (module.js:470:15)
npm verb stack     at Function.Module._load (module.js:418:25)
npm verb stack     at Module.require (module.js:498:17)
npm verb stack     at require (internal/module.js:20:19)
npm verb stack     at Object.<anonymous> (C:\Users\mbart\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcorp\move\move.js:7:12)
npm verb stack     at Module._compile (module.js:571:32)
npm verb stack     at Object.Module._extensions..js (module.js:580:10)
npm verb stack     at Module.load (module.js:488:32)
npm verb stack     at tryModuleLoad (module.js:447:12)
npm verb stack     at Function.Module._load (module.js:439:3)
npm verb cwd C:\WINDOWS\system32
npm verb Windows_NT 10.0.14393
npm verb argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\mbart\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "qawgehrsfgrw" "--verbose"
npm verb node v7.7.2
npm verb npm  v4.4.2
npm ERR! code MODULE_NOT_FOUND
npm ERR! Cannot find module '@npmcorp/copy'
npm verb exit [ 1, true ]

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\mbart\AppData\Roaming\npm-cache\_logs\2017-03-14T18_26_37_433Z-debug.log
````

(I know I'm installing gibberish, the same error happens with a real package such as grunt and parsoid)
Log file:

````
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\mbart\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   'qawgehrsfgrw' ]
2 info using npm@4.4.2
3 info using node@v7.7.2
4 verbose stack Error: Cannot find module '@npmcorp/copy'
4 verbose stack     at Function.Module._resolveFilename (module.js:470:15)
4 verbose stack     at Function.Module._load (module.js:418:25)
4 verbose stack     at Module.require (module.js:498:17)
4 verbose stack     at require (internal/module.js:20:19)
4 verbose stack     at Object.<anonymous> (C:\Users\mbart\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcorp\move\move.js:7:12)
4 verbose stack     at Module._compile (module.js:571:32)
4 verbose stack     at Object.Module._extensions..js (module.js:580:10)
4 verbose stack     at Module.load (module.js:488:32)
4 verbose stack     at tryModuleLoad (module.js:447:12)
4 verbose stack     at Function.Module._load (module.js:439:3)
5 verbose cwd C:\WINDOWS\system32
6 verbose Windows_NT 10.0.14393
7 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\mbart\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "qawgehrsfgrw"
8 verbose node v7.7.2
9 verbose npm  v4.4.2
10 error code MODULE_NOT_FOUND
11 error Cannot find module '@npmcorp/copy'
12 verbose exit [ 1, true ]
````

Working on Windows 10 with a fresh install of Node.js and npm with admin privileges.

`npm -v` returns `4.4.2`

Installing '@npmcorp/copy' does nothing, it returns the exact same error.

**Comment by Quentame** [comment]
I got exactly the same issue, on Mac OS 10.9 just right now:

When I want to install/update : 
```java
npm ERR! code MODULE_NOT_FOUND
npm ERR! Cannot find module '@npmcorp/move'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/PQ/.npm/_logs/2017-03-15T00_26_59_611Z-debug.log
```

Log file:
```java
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/Cellar/node/5.4.0/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'install',
1 verbose cli   'csscomb',
1 verbose cli   '-g' ]
2 info using npm@4.4.2
3 info using node@v5.4.0
4 verbose stack Error: Cannot find module '@npmcorp/move'
4 verbose stack     at Function.Module._resolveFilename (module.js:326:15)
4 verbose stack     at Function.Module._load (module.js:277:25)
4 verbose stack     at Module.require (module.js:354:17)
4 verbose stack     at require (internal/module.js:12:17)
4 verbose stack     at Object.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/move.js:5:12)
4 verbose stack     at Module._compile (module.js:398:26)
4 verbose stack     at Object.Module._extensions..js (module.js:405:10)
4 verbose stack     at Module.load (module.js:344:32)
4 verbose stack     at Function.Module._load (module.js:301:12)
4 verbose stack     at Module.require (module.js:354:17)
4 verbose stack     at require (internal/module.js:12:17)
5 verbose cwd /Users/PQ/Documents/contributing/jQuery.mmenu
6 verbose Darwin 13.4.0
7 verbose argv "/usr/local/Cellar/node/5.4.0/bin/node" "/usr/local/bin/npm" "install" "csscomb" "-g"
8 verbose node v5.4.0
9 verbose npm  v4.4.2
10 error code MODULE_NOT_FOUND
11 error Cannot find module '@npmcorp/move'
12 verbose exit [ 1, true ]
```

It happens after ```sudo npm update -g``` 😞 

Stack trace : 
```java
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN install:mkdirp@0.5.1 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/mkdirp/bin/cmd.js'
npm WARN install:nopt@3.0.6 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/nopt/bin/nopt.js'
npm WARN install:opener@1.4.1 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/opener/opener.js'
npm WARN install:sshpk@1.9.2 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/request/node_modules/http-signature/node_modules/sshpk/bin/sshpk-sign'
npm WARN install:node-uuid@1.4.7 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/request/node_modules/node-uuid/bin/uuid'
npm WARN install:semver@5.1.0 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/semver/bin/semver'
npm WARN install:har-validator@2.0.6 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/request/node_modules/har-validator/bin/har-validator'
npm WARN install:which@1.2.11 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/which/bin/which'
npm WARN install:rimraf@2.5.4 ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/cordova/node_modules/npm/node_modules/rimraf/bin.js'
/usr/local/bin/budo -> /usr/local/lib/node_modules/budo/bin/cmd.js
/usr/local/bin/npm -> /usr/local/lib/node_modules/npm/bin/npm-cli.js

> fsevents@1.1.1 install /usr/local/lib/node_modules/budo/node_modules/fsevents
> node install

events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: spawn node-pre-gyp ENOENT
    at exports._errnoException (util.js:856:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:478:9)
    at process._tickCallback (node.js:392:17)
    at Function.Module.runMain (module.js:432:11)
    at startup (node.js:141:18)
    at node.js:1003:3
npm WARN install:fsevents@1.1.1 fsevents@1.1.1 install: `node install`
npm WARN install:fsevents@1.1.1 Exit status 1
- acorn@3.3.0 node_modules/budo/node_modules/detective/node_modules/acorn
- abbrev@1.1.0 node_modules/budo/node_modules/fsevents/node_modules/abbrev
- ansi-regex@2.1.1 node_modules/budo/node_modules/fsevents/node_modules/ansi-regex

... [long list]

- read-installed@4.0.3 node_modules/npm/node_modules/read-installed
- read-package-tree@5.1.5 node_modules/npm/node_modules/read-package-tree
- write-file-atomic@1.3.1 node_modules/npm/node_modules/write-file-atomic
/usr/local/lib
├── @npmcorp/move@1.0.0
├── abbrev@1.1.0
├── ansi-regex@2.1.1

... [some more]

├── validate-npm-package-name@3.0.0
└── write-file-atomic@1.3.1

npm WARN EBUNDLEOVERRIDE Replacing bundled npm > @npmcorp with new installed version
```

Hope that helps 😉

**Comment by zkat** [comment]
This is fixed by #16040 -- y'all will need to stick to `npm@latest` instead of updating to `npm@next`. And you may need to purge your npm installation (by manually removing the `npm` directory in your installation) because the node installer often neglects to remove files.

**Comment by crishna369** [comment]
I am also getting the same error while installing any packages

**Comment by zkat** [comment]
@crishna369 this error happens in exactly one version of npm, which was never distributed with node. If you are seeing this actual error, it is likely that you're mistaken about which npm you're actually running. Check your path, and remove the offending npm installation.

**Comment by crishna369** [comment]
How can I find the path of current npm and remove the additional npm installation? Please help me!

**Comment by zkat** [comment]
@crishna369 that's something you're gonna have to look for help with elsewhere. I can't really help you with this.

**Comment by crishna369** [comment]
I installed node 6.9.5 using nvm and now it is working. Anyway thank you for you help.

**Comment by mbartelsm** [comment]
Thanks, I used `npm@next` following a setup guide from MediaWiki for Parsoid. I will comment the issue there to have it fixed and avoid further instances of this issue.

**Comment by zkat** [comment]
Fix released.

@mbartelsm setup guides should probably not use `npm@next`. That is very much a beta channel, and it's our canary in the gold mine for broken versions.

Closing this because fixed in 4.4.3

**Comment by DmitryGonchar** [comment]
Just one addition to the 

> you may need to purge your npm installation (by manually removing the npm directory in your installation) 

At least on Windows you need to remove `npm` and `npm-cache` folders in `C:\Users\UserName\AppData\Roaming`, and _not_ in the Program Files where node.js is installed (the latter caused an error that "Cannot find module npm-cli")
